src/geventhttpclient/utils.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import six
+
+if six.PY3:
+    from collections.abc import Mapping
+else:
+    from collections import Mapping
+basestring = (str, bytes)
+
+
+def to_key_val_list(value):
+    """Take an object and test to see if it can be represented as a
+    dictionary. If it can be, return a list of tuples, e.g.,
+
+    ::
+
+        >>> to_key_val_list([('key', 'val')])
+        [('key', 'val')]
+        >>> to_key_val_list({'key': 'val'})
+        [('key', 'val')]
+        >>> to_key_val_list('string')
+        Traceback (most recent call last):
+        ...
+        ValueError: cannot encode objects that are not 2-tuples
+
+    :rtype: list
+    """
+    if value is None:
+        return None
```

Comment:
you're never calling this function with None, and thrown an exception makes more sense if someone actually DID call it that way, so maybe remove this?

Reply:
I just realised there is an exact same method in url.py I ll use that instead